### PR TITLE
Deployment plans search

### LINKED ISSUE #5990: Deployment plans search
I just noticed that `DeploymentPlans` did not get the new look for the search filters that was added everywhere else some time ago

deployment plans

//cc @agriffard

### PR DESCRIPTION
**Pull request by agriffard**
Select all and bulk actions (Delete):

image

Fixes #5990

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs [hunk 1]
```diff
@@ -105,9 +108,49 @@ public async Task<IActionResult> Index(DeploymentPlanIndexOptions options, Pager
...
+                switch (options.BulkAction)
+                {
+                    case DeploymentPlansBulkAction.None:
+                        break;
+                    case DeploymentPlansBulkAction.Delete:
+                        foreach (var item in checkedItems)
+                        {
+                            _session.Delete(item);
+
+                            _notifier.Success(H["Deplyment plan {0} successfully deleted.", item.Name]);
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs:143]:
```suggestion
                            _notifier.Success(H["Deployment plan {0} successfully deleted.", item.Name]);
```

File: src/OrchardCore.Modules/OrchardCore.Deployment/Views/_ViewImports.cshtml [hunk 2]
```diff
@@ -6,3 +6,6 @@
 @addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
 @addTagHelper *, OrchardCore.DisplayManagement
 @addTagHelper *, OrchardCore.ResourceManagement
+
+@using Microsoft.Extensions.Localization;
+@using Microsoft.AspNetCore.Mvc.Localization;
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Views/_ViewImports.cshtml:11]: newline

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs [hunk 3]
```diff
@@ -105,9 +108,49 @@ public async Task<IActionResult> Index(DeploymentPlanIndexOptions options, Pager
...
+                        }
+                        break;
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs:145]: I think the notification for most of the others is done once, after the loop (without the name). Not bothered either way

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs [hunk 4]
```diff
@@ -105,9 +108,49 @@ public async Task<IActionResult> Index(DeploymentPlanIndexOptions options, Pager
...
+
+                            _notifier.Success(H["Deployment plan {0} successfully deleted.", item.Name]);
+                        }
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            return RedirectToAction("Index");
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs:151]: Can we use `nameof (Index)` everywhere

File: src/OrchardCore.Modules/OrchardCore.Deployment/Indexes/DeploymentPlanIndex.cs [hunk 5]
```diff
@@ -4,6 +4,7 @@ namespace OrchardCore.Deployment.Indexes
 {
     public class DeploymentPlanIndex : MapIndex
     {
+        public string DocumentId { get; set; }
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Indexes/DeploymentPlanIndex.cs:7]: Add new line after

File: src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs [hunk 6]
```diff
@@ -105,9 +108,49 @@ public async Task<IActionResult> Index(DeploymentPlanIndexOptions options, Pager
...
             };
 
+            model.Options.DeploymentPlansBulkAction = new List<SelectListItem>() {
+                new SelectListItem() { Text = S["Delete"], Value = nameof(DeploymentPlansBulkAction.Delete) }
+            };
+
             return View(model);
         }
 
+        [HttpPost, ActionName("Index")]
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs:118]: Also `"Index"` in attributes can be replaced too

Comment (agriffard) [reply src/OrchardCore.Modules/OrchardCore.Deployment/Controllers/DeploymentPlanController.cs]: ok

**Pull request merged**